src/main/java/org/openrewrite/java/migrate/guava/NoGuavaPrimitiveAsList.java

```diff
@@ -0,0 +1,88 @@
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.template.internal.AbstractRefasterJavaVisitor;
+import org.openrewrite.java.tree.J;
+
+public class NoGuavaPrimitiveAsList extends Recipe {
+
+    static final MethodMatcher METHOD_MATCHER = new MethodMatcher("com.google.common.primitives.* asList(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "`com.google.common.primitives asList` to `Arrays.asList`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Migrate from Guava `com.google.common.primitives asList` to `Arrays.asList`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
+
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
+                if (METHOD_MATCHER.matches(elem)) {
+                    maybeRemoveImport("com.google.common.primitives.*");
+                    maybeRemoveImport("com.google.common.primitives.Booleans");
+                    maybeRemoveImport("com.google.common.primitives.Chars");
+                    maybeRemoveImport("com.google.common.primitives.Doubles");
+                    maybeRemoveImport("com.google.common.primitives.Floats");
+                    maybeRemoveImport("com.google.common.primitives.Longs");
+                    maybeRemoveImport("com.google.common.primitives.Ints");
+                    maybeRemoveImport("com.google.common.primitives.Shorts");
+                    maybeRemoveImport("com.google.common.primitives.Bytes");
+                    maybeAddImport("java.util.Arrays");
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Arrays.asList(");
+                    for (int i = 0; i < elem.getArguments().size(); i++) {
+                        sb.append("#{any()}");
+                        if (i != elem.getArguments().size() - 1) {
+                            sb.append(", ");
+                        }
+                    }
+                    sb.append(")");
+                    return JavaTemplate
+                            .builder(sb.toString())
+                            .imports("java.util.Arrays")
+                            .build()
+                            .apply(getCursor(), elem.getCoordinates().replace(), elem.getArguments().toArray());
+                }
+                return super.visitMethodInvocation(elem, ctx);
+            }
+
+        };
+        return Preconditions.check(
+                Preconditions.and(
+                        new UsesType<>("java.util.List", true),
+                        new UsesMethod<>("com.google.common.primitives.* asList(..)")
+                ),
```

Comment:
The second one implies the first right? No need to check both in that case I think.
```suggestion
                new UsesMethod<>("com.google.common.primitives.* asList(..)"),
```

Reply:
Sounds fair!